opslevel/resource_opslevel_service.go

```diff
@@ -247,13 +247,32 @@ func (r *ServiceResource) Create(ctx context.Context, req resource.CreateRequest
 			}
 		}
 	}
-	service, err = r.client.GetService(opslevel.ID(service.Id))
+	service, err = r.client.GetService(service.Id)
 	if err != nil {
 		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to get service after creation, got error: %s", err))
 		return
 	}
 
 	stateModel, diags := NewServiceResourceModel(ctx, *service)
+	resp.Diagnostics.Append(diags...)
+
+	// after creating resource model, differentiate between a basic field being unset (null) vs empty string ("")
+	// TODO: find a way to do this without repeating code
```

Comment:
Oh i assumed we'd do this in the `read` function or in the `NewServiceResourceModel` but you'd have to pass in the plan to it so it can check before it sets the value to the `stateModel`

Reply:
cc @davidbloss would save us many lines of code to use the `planModel` as part of `NewResourceModel()` functions

Reply:
Go for it 👍